### ISSUE TITLE
False `custom_element_props_identifier` errors with no custom elements

### ISSUE THREAD

**Issue by kyoshino** [sveltejs/eslint-plugin-svelte#1241]
### Describe the bug

After sveltejs/svelte#16003 landed in [svelte@5.33.4](https://github.com/sveltejs/svelte/releases/tag/svelte%405.33.4), I started getting `custom_element_props_identifier` errors with a `rest` prop even though my project doesn’t use custom elements. These errors come from ESLint with `eslint-plugin-svelte` installed, but not from `svelte-check`.

I could silence the errors by adding `svelte/valid-compile: off` to the ESLint config, but something is going wrong on the Svelte side. It may be a bug in `eslint-plugin-svelte`, but I’m not sure.

### Reproduction

Repo: https://github.com/sveltia/sveltia-cms

One of the errors is raised here: [`image.svelte`](https://github.com/sveltia/sveltia-cms/blob/main/src/lib/components/assets/shared/image.svelte#L33)

A minimum reproduction:

```svelte
<script>
  let { ...rest } = $props();
</script>
```

### Logs

```shell
$ eslint .

/Users/Kohei/Sites/sveltia-cms/src/lib/components/assets/shared/asset-preview.svelte
  45:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/assets/shared/image.svelte
  33:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/assets/shared/video.svelte
  30:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/common/list-container.svelte
  18:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/common/listing-grid.svelte
  25:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/common/page-container-main-area.svelte
  21:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/common/page-container.svelte
  21:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

/Users/Kohei/Sites/sveltia-cms/src/lib/components/common/page-toolbar/view-switcher.svelte
  23:5  error  Using a rest element or a non-destructured declaration with `$props()` means that Svelte can't infer what properties to expose when creating a custom element. Consider destructuring all the props or explicitly specifying the `customElement.props` option.
https://svelte.dev/e/custom_element_props_identifier(custom_element_props_identifier)  svelte/valid-compile

✖ 8 problems (8 errors, 0 warnings)
```

### System Info

```shell
-
```

### Severity

annoyance

**Comment by Rich-Harris** [comment]
Transferred issue to `eslint-plugin-svelte`

**Comment by kyoshino** [comment]
I cannot upgrade `eslint-plugin-svelte` to v3 because I’m waiting for `eslint-config-airbnb-base` to support ESLint v9, which requires the new flat config format, but thanks for the fix anyway! 👏🏼
